cdc/log.cc

```diff
@@ -401,8 +401,11 @@ static schema_ptr create_log_schema(const schema& s, std::optional<utils::UUID>
             auto type = column.type;
             if (is_data_col) {
                 type = visit(*type, make_visitor(
-                    // lists are represented as map<timeuuid, value_type>. Otherwise we cannot express delta
+                    // non-frozen lists are represented as map<timeuuid, value_type>. Otherwise we cannot express delta
                     [] (const list_type_impl& type) -> data_type {
+                        if (type.is_atomic()) {
+                            return type.freeze();
```

Comment:
You can probably just return type since it's already frozen, no?

Reply:
also the `type = type->freeze()` line (416) is redundant. But nvm.

Reply:
@haaawk Changed `freeze()` to `shared_from_this()`.

Reply:
maybe return column.type instead?

Reply:
or rename `type` to `typ`, catch `&type`, and return it. (you can even `std::move` it).

Reply:
Yes. That would work too.

Reply:
@haaawk @kbr- How about now? I took a slightly different approach, and removed redundant freezing by the way.

Reply:
Looks great now.